sklearn/kernel_approximation.py

```diff
@@ -32,6 +36,37 @@
 )
 
 
+# Utility Functions
+def _return_float_dtype(X, Y):
+    """
+    1. If dtype of X and Y is float32, then dtype float32 is returned.
+    2. Else dtype float is returned.
+    """
+    if not sp.issparse(X) and not isinstance(X, np.ndarray):
+        X = np.asarray(X)
+    if Y is None:
+        Y_dtype = X.dtype
+    elif not sp.issparse(Y) and not isinstance(Y, np.ndarray):
+        Y = np.asarray(Y)
+        Y_dtype = Y.dtype
+    else:
+        Y_dtype = Y.dtype
+    if X.dtype == Y_dtype == np.float32:
+        dtype = np.float32
+    else:
+        dtype = float
+    return X, Y, dtype
+
+
+def _find_floating_dtype_allow_sparse(X, Y, xp=None):
```

Comment:
Instead of duplicating the `_return_float_dtype` function of `sklearn/metrics/pairwise.py` here, I would rather move `_find_floating_dtype_allow_sparse` to `sklearn/metrics/pairwise.py` because it is only used once there.